edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.Http.Test/DeviceScopeControllerTest.cs

```diff
@@ -76,27 +76,131 @@ public async Task NestedDeviceScopeRoundtripTest()
+
+            Assert.Equal((int)HttpStatusCode.OK, controller.HttpContext.Response.StatusCode);
+            Assert.Equal(responseExpectedJson, responseActualJson);
+        }
+
+        [Fact]
+        public async Task GetDeviceOnBehalfOf()
+        {
+            // Setup ServiceIdentity results
+            string parentEdgeId = "edge1";
+            string childEdgeId = "edge2";
+            string deviceId = "device1";
+            string moduleId = "module1";
+            string deviceScope = "deviceScope1";
+            string parentScope = "parentScope1";
+            string generationId = "generation1";
+            string primaryKey = "t3LtII3CppvtVqycKp9bo043vCEgWbGBJAzXZNmoBXo=";
+            string secondaryKey = "kT4ac4PpH5UY0vA1JpLQWOu2yG6qKoqwvzee3j1Z3bA=";
+            var authentication = new ServiceAuthentication(new SymmetricKeyAuthentication(primaryKey, secondaryKey));
+            var resultDeviceIdentity = new ServiceIdentity(deviceId, null, deviceScope, new List<string>() { parentScope }, generationId, Enumerable.Empty<string>(), authentication, ServiceIdentityStatus.Enabled);
+            var resultModuleIdentity = new ServiceIdentity(deviceId, moduleId, null, new List<string>() { deviceScope }, generationId, Enumerable.Empty<string>(), authentication, ServiceIdentityStatus.Enabled);
+            var resultIdentities = new List<ServiceIdentity>() { resultDeviceIdentity, resultModuleIdentity };
+            var authChainMapping = new Dictionary<string, string>();
+            authChainMapping.Add(deviceId, "device1;edge2;edge1;edgeroot");
+            var controller = MakeController(childEdgeId, resultIdentities, authChainMapping);
+
+            // Act
+            var request = new IdentityOnBehalfOfRequest(deviceId, null, "device1;edge2;edge1");
+            await controller.GetDeviceAndModuleOnBehalfOf(parentEdgeId, "$edgeHub", request);
+
+            // Verify EdgeHub result types
+            var expectedAuth = new AuthenticationMechanism() { SymmetricKey = new SymmetricKey() { PrimaryKey = primaryKey, SecondaryKey = secondaryKey } };
+            var expectedDeviceIdentities = new List<EdgeHubScopeDevice>() { new EdgeHubScopeDevice(deviceId, generationId, DeviceStatus.Enabled, expectedAuth, new DeviceCapabilities(), deviceScope, new List<string> { parentScope }) };
+            var responseExpected = new EdgeHubScopeResult() { Devices = expectedDeviceIdentities, Modules = new List<EdgeHubScopeModule>() };
+            var responseExpectedJson = JsonConvert.SerializeObject(responseExpected);
+
+            var responseActualBytes = GetResponseBodyBytes(controller);
+            var responseActualJson = Encoding.UTF8.GetString(responseActualBytes);
+
+            Assert.Equal((int)HttpStatusCode.OK, controller.HttpContext.Response.StatusCode);
+            Assert.Equal(responseExpectedJson, responseActualJson);
+        }
+
+        [Fact]
+        public void ValidateChainTest()
```

Comment:
I think it is better to move this test to another file "AuthChainHelpersTest.cs"